src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/CpsProjectSystemReferenceReader.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.References;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.ProjectManagement;
+using NuGet.ProjectModel;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Reference reader implementation for the CPS project system in the integrated development environment (IDE).
+    /// </summary>
+    internal class CpsProjectSystemReferenceReader
+        : IProjectSystemReferencesReader
+    {
+        private readonly IVsProjectAdapter _vsProjectAdapter;
+        private readonly IVsProjectThreadingService _threadingService;
+        private readonly AsyncLazy<ConfiguredProject> _configuredProject;
+
+        public CpsProjectSystemReferenceReader(
+            IVsProjectAdapter vsProjectAdapter,
+            IVsProjectThreadingService threadingService)
+        {
+            Assumes.Present(vsProjectAdapter);
+            Assumes.Present(threadingService);
+
+            _vsProjectAdapter = vsProjectAdapter;
+            _threadingService = threadingService;
+            _configuredProject = new AsyncLazy<ConfiguredProject>(async () =>
+            {
+                await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                var context = _vsProjectAdapter.Project as IVsBrowseObjectContext;
+                if (context == null)
+                {
+                    // VC implements this on their DTE.Project.Object
+                    context = _vsProjectAdapter.Project.Object as IVsBrowseObjectContext;
+                }
+                return context?.ConfiguredProject;
+            }, threadingService.JoinableTaskFactory);
+        }
+
+        public async Task<IEnumerable<ProjectRestoreReference>> GetProjectReferencesAsync(
+            Common.ILogger logger, CancellationToken _)
+        {
+            var project = await _configuredProject.GetValueAsync();
+            IBuildDependencyProjectReferencesService service = project.Services.ProjectReferences;
+
+            if (service == null)
+            {
+                return Enumerable.Empty<ProjectRestoreReference>();
+            }
+            var results = new List<ProjectRestoreReference>();
+            var hasMissingReferences = false;
+
+            foreach (IUnresolvedBuildDependencyProjectReference projectReference in await service.GetUnresolvedReferencesAsync())
+            {
+                try
+                {
+                    if (!await projectReference.GetReferenceOutputAssemblyAsync())
```

Comment:
This should be the reverse, right?

Reply:
yep, should be true. 

GetFullPathAsync is not returning a value right now.  
It's returning null. 

I'll retest tomorrow.

Reply:
EvaluatedIncludeAsFullPath property seems to work.

Reply:
Looks like GetFullPathAsync is returning the resolved path otherwise, null. Your latest change is correct.

Reply:
`EvaluatedIncludeAsFullPath` property seems to work.